astro.config.mjs

```diff
@@ -4,16 +4,36 @@ import sitemap from "@astrojs/sitemap";
 import tailwind from "@astrojs/tailwind";
 import sectionize from "@hbsnow/rehype-sectionize";
 import icon from "astro-icon";
-import pagefind from "astro-pagefind";
 import { defineConfig } from "astro/config";
+import {
+  addEmbeddedArticles,
+  getArticleMatches,
+} from "./src/utils/relatedArticles";
+
+// Custom Astro integration
+function buildStart() {
+  return {
+    name: "my-build-start",
+    hooks: {
+      "astro:build:start": async () => {
+        const branch = process.env.CF_PAGES_BRANCH || "unknown";
+        if (branch === "main") {
+          await addEmbeddedArticles();
+        }
+
+        await getArticleMatches();
+
+        // await addEmbeddedArticles();
```

Comment:
remove commented code

Reply:
Astro does not support custom build flags, so I'll just remove the comments.